inception-ui-external-search/src/main/java/de/tudarmstadt/ukp/inception/app/ui/externalsearch/SearchPage.java

```diff
@@ -334,6 +336,15 @@ public ResultRowView(String id, long rowNumber, IModel<ExternalSearchResult> mod
             add(new LambdaAjaxLink("importLink", _target -> actionImportDocument(_target, result))
                     .add(visibleWhen(() -> 
                         !documentService.existsSourceDocument(project, documentTitle))));
+            add(new LambdaAjaxLink("openLink", _target -> {
+                PageParameters pageParameters = new PageParameters()
+                    .add(WebAnnoConst.PAGE_PARAM_PROJECT_ID, project.getId())
+                    .add(WebAnnoConst.PAGE_PARAM_DOCUMENT_ID,
+                        documentService.getSourceDocument(project, documentTitle).getId())
```

Comment:
Does this return the same thing as `result.getDocumentId()`?

Reply:
The ID in the result is an ID used by the external document repository. The SourceDocument ID is internal to INCEpTION and generated only upon import. While using the repository ID as the document title works, it feels a bit fragile and is probably something that needs to be refactored later on. It might be a good idea to introduce an `externalId` column in the `SourceDocument` class which can be used to carry the original ID a document had in an external repository.

Reply:
Probably would need two columns: 
* `source` to identify where the document came from (maybe a string to be flexible)
* `sourceId` to uniquely identify the document within the original source (best also a string)

Reply:
Created issue #824 for this refactoring